monitor/launch/launcher.go

```diff
@@ -75,42 +75,52 @@ func (nm *NodeMonitor) GetPid() int {
 	return nm.GetProcess().Pid
 }
 
-// Run starts the node monitor.
-func (nm *NodeMonitor) Run(sockPath, bpfRoot string) {
-	nm.SetTarget(targetName)
-	for {
-		os.Remove(sockPath)
-		if err := syscall.Mkfifo(sockPath, 0600); err != nil {
-			log.WithError(err).Fatalf("Unable to create named pipe %s", sockPath)
-			time.Sleep(time.Duration(5) * time.Second)
-		}
+func (nm *NodeMonitor) run(sockPath, bpfRoot string) error {
+	os.Remove(sockPath)
+	if err := syscall.Mkfifo(sockPath, 0600); err != nil {
+		return fmt.Errorf("Unable to create named pipe %s: %s", sockPath, err)
+	}
+
+	pipe, err := os.OpenFile(sockPath, os.O_RDWR, 0600)
+	if err != nil {
+		return fmt.Errorf("Unable to open named pipe for writing: %s", err)
+	}
+
+	defer pipe.Close()
+
+	nm.Mutex.Lock()
+	nm.pipe = pipe
+	nm.Mutex.Unlock()
+
+	nm.Launcher.SetArgs([]string{"--bpf-root", bpfRoot})
+	nm.Launcher.Run()
 
-		pipe, err := os.OpenFile(sockPath, os.O_RDWR, 0600)
+	r := bufio.NewReader(nm.GetStdout())
+	for nm.GetProcess() != nil {
+		l, err := r.ReadBytes('\n') // this is a blocking read
 		if err != nil {
-			log.WithError(err).Fatal("Unable to open named pipe for writing")
-			time.Sleep(time.Duration(5) * time.Second)
+			return fmt.Errorf("Unable to read stdout from monitor: %s", err)
 		}
 
-		nm.Mutex.Lock()
-		nm.pipe = pipe
-		nm.Mutex.Unlock()
-
-		nm.Launcher.SetArgs([]string{"--bpf-root", bpfRoot})
-		nm.Launcher.Run()
-
-		r := bufio.NewReader(nm.GetStdout())
-		for nm.GetProcess() != nil {
-			l, _ := r.ReadBytes('\n') // this is a blocking read
-			var tmp *models.MonitorStatus
-			if err := json.Unmarshal(l, &tmp); err != nil {
-				continue
-			}
-			nm.setState(tmp)
+		var tmp *models.MonitorStatus
+		if err := json.Unmarshal(l, &tmp); err != nil {
+			return fmt.Errorf("Unable to unmarshal stdout from monitor: %s", err)
 		}
 
-		pipe.Close()
+		nm.setState(tmp)
+	}
+
+	return nil
+}
+
+// Run starts the node monitor.
```

Comment:
Since you're editing things anyway, could you also add here that `Run` does not return and that it retries setting up the pipe and reading from node-monitor forever.